framework/src/service/ServiceReferenceBasePrivate.cpp

```diff
@@ -103,64 +114,51 @@ InterfaceMapConstPtr ServiceReferenceBasePrivate::GetPrototypeService(const Bund
 
 std::shared_ptr<void> ServiceReferenceBasePrivate::GetService(BundlePrivate* bundle)
 {
-  auto s = ExtractInterface(GetServiceInterfaceMap(bundle), interfaceId);
-  if (!s)
-  {
-    registration->Lock(), --registration->dependents[bundle];
-  }
-  return s;
+  return ExtractInterface(GetServiceInterfaceMap(bundle), interfaceId);
```

Comment:
good catch. The code changes should guarantee that ```nullptr``` is never passed to ```ExtractInterface``` (in all failure modes, it should be a default constructed ```InterfaceMapConstPtr```). Nevertheless its a good idea to ensure it never can dereference a ```nullptr```. I'll fix it.